### PR TITLE
Adding docs for how to run importable workflow

### PR DESCRIPTION
**Pull request by chatton**
## Description

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -352,3 +353,45 @@ json matrix files under .github/compatibility-test-matrices and is equivalent to
   ```
 
   This issue doesn't seem to occur on other operating systems.
+
+## Importable Workflow
+
+This repo contains an [importable workflow](https://github.com/cosmos/ibc-go/blob/bc963bcfd115a0e06b8196b114496db5ea011247/.github/workflows/e2e-compatibility-workflow-call.yaml) that can be used from any other to test chain upgrades. The workflow
```
Comment (damiannolan) [inline e2e/README.md:359]: nit on wording 

```suggestion
This repository contains an [importable workflow](https://github.com/cosmos/ibc-go/blob/bc963bcfd115a0e06b8196b114496db5ea011247/.github/workflows/e2e-compatibility-workflow-call.yaml) that can be used from any other repository to test chain upgrades. The workflow
```

**Review by damiannolan — APPROVED** [review]
Looks good to me! :)

File: e2e/README.md [hunk 2]
```diff
@@ -352,3 +353,45 @@ json matrix files under .github/compatibility-test-matrices and is equivalent to
...
+You can refer to [this example](https://github.com/cosmos/ibc-go/blob/2933906d1ed25ae6dce7b7d93aa429dfa94c5a23/.github/workflows/e2e-upgrade.yaml#L9-L19) when including this workflow in your repo.
+
+The referenced job will do the following:
+
+- Create two chains using the image found at `ghcr.io/cosmos/ibc-go-simd:v4.3.0`.
+- Perform IBC transfers verifying core functionality.
+- Upgrade chain A to `ghcr.io/cosmos/ibc-go-simd:v5.1.0` by executing a governance proposal and using the plan name `normal upgrade`.
+- Perform additional IBC transfers and verifies the upgrade and migrations ran successfully.
+
+> Note: The plan name will always be specific to your chain. In this instance `normal upgrade` is referring to [this upgrade handler](https://github.com/cosmos/ibc-go/blob/e9bc0bac38e84e1380ec08552cae15821143a6b6/testing/simapp/app.go#L923)
```
Comment (colin-axner) [inline e2e/README.md:382]: Should we note that they need to add an upgrade handler to their binary?

**Review by colin-axner — APPROVED** [review]
Wonderful! :) 

This is really cool. Hopefully will catch some issues early on for chains

File: e2e/README.md (hunk 2 above)
Comment (chatton) [reply e2e/README.md:382]: No harm adding as a prerequisite to have an upgrade handler registered!

**Pull request merged**